source/slang/slang-ir.h

```diff
@@ -2322,6 +2323,25 @@ struct IRModule : RefObject
     {
         return m_containerPool;
     }
+
+    // Raytracing variables/attributes managment functions
+    bool checkedForRayVariables = false;
```

Comment:
These functions should belong to a separate pass instead of in `IRModule`.

Reply:
There is a small problem with moving the functions+var's of this pass; even though for-now it was decided to drop HLSL support for GLSL raytracing functions, if it ever is brought back (or CUDA, or other), the exact same function & pass will need to be shared between all of the API's, in which I don't exactly know where to store the data which is needed for the final pass (code gen/emit'ting) such that all can access the data.

the options are to:
1. keep it where it is for the future
or
2. GLSL will only go to SPIR-V and GLSL, therefore move the functions+var's inside the SPIR-V emitter

Reply:
My understanding is that there is no need to store that data explicitly on the IR, since you can construct the dictionary from the decorations, no?

Reply:
You can write a separate function, called:
```
Dictionary<int, IRInst*> createMapFromVulkanLocationToPayloadVar(IRModule*);
```
And define it in a separate header like `slang-ir-vulkan-location-to-payload.h` and implement it in a cpp.

Then anything that needs this can just include the header file and call the function to get the map constructed from IR.

Reply:
👍I will move the logic outside the Module and cache the data in the code pass structure that needs the data (SourceEmitterBase)

Reply:
Sounds good!

Reply:
The general rule is to do a little translation work as possible at emit. Emit should stay as direct as possible from IR to target code, because it is far easier to maintain separate small and well-defined passes than to maintain a big emit pass that does 100 different things, especially when some of those things may have consequences on each other.

Doing a pass scanning through all the opcodes is fine.